### PR TITLE
Refactor ModelLoader to move away from the common pool executor.

### PR DESCRIPTION
**Pull request by regadas**
`zoltar`  should avoid the use of a default `Executor` and it should not assume one. Instead we should allow the user to provide their one! With this we keep a clean and simple api that allows several use cases and avoids the common concurrency pitfals associated with the use of the default `ForkJoinPool.commonPool()`.

### CODE REVIEW

File: zoltar-core/src/main/java/com/spotify/zoltar/ModelLoader.java [hunk 1]
```diff
@@ -48,19 +49,141 @@
...
+
+  }
+
+  /**
+   * ConsLoader is a constant {@link ModelLoader}.
+   *
+   * @param <M> Model instance type.
+   */
+  @FunctionalInterface
+  interface ConsLoader<M extends Model<?>> extends ModelLoader<M> {
```
Comment (brianmartin) [inline zoltar-core/src/main/java/com/spotify/zoltar/ModelLoader.java]: Why not ``ConstantLoader`? ditto for the `cons` function, why not `constant`?

File: zoltar-tests/src/test/java/com/spotify/zoltar/ModelLoaderTest.java [hunk 2]
```diff
@@ -52,35 +54,27 @@ public void close() throws Exception {
...
-    assertThat(loader.get().toCompletableFuture().isDone(), is(true));
-  }
-
-  @Test
-  public void preloadTimeout() {
-    final ModelLoader<DummyModel> loader = ModelLoader
-        .lift(() -> {
-          Thread.sleep(Duration.ofSeconds(10).toMillis());
+        .load(() -> {
+          Thread.sleep(Duration.ofMillis(500).toMillis());
```
Comment (brianmartin) [inline zoltar-tests/src/test/java/com/spotify/zoltar/ModelLoaderTest.java]: Any reason this shouldn't be something much smaller? Like `5ms`?

**Comment by regadas** [comment]
@brianmartin 

> Is it a bit strange to have the non-nullable executor argument following nullable arguments?

imho it's ok to have it like that and it's a consistent api.

> Should we still use ForkJoinPool.commonPool by default?

I thought about it! The reason why i removed it was to make the api clearer, concise and because it should not make assumption on which `Executor` to use. Providing a default one can lead to confusion and to some concurrency pitfals.

The reason above and the possible use cases it allows like the possibility of supplying an instrumented `Executor` were the reason for this refactor.

File: zoltar-core/src/main/java/com/spotify/zoltar/ModelLoader.java (hunk 1 above)
Comment (regadas) [reply zoltar-core/src/main/java/com/spotify/zoltar/ModelLoader.java]: functional lingo

File: zoltar-tests/src/test/java/com/spotify/zoltar/ModelLoaderTest.java (hunk 2 above)
Comment (regadas) [reply zoltar-tests/src/test/java/com/spotify/zoltar/ModelLoaderTest.java]: yeah 😄

File: zoltar-core/src/main/java/com/spotify/zoltar/ModelLoader.java (hunk 1 above)
Comment (brianmartin) [reply zoltar-core/src/main/java/com/spotify/zoltar/ModelLoader.java]: Lisp?

Comment (regadas) [reply zoltar-core/src/main/java/com/spotify/zoltar/ModelLoader.java]: yeah and you can see it in haskell and scala as well.

**Pull request merged**